app/models/katello/candlepin/content.rb

```diff
@@ -79,6 +79,11 @@ def build_repository
         repository
       end
 
+      def get_certificate_and_key(product, modified_product_ids = [])
+        product = Product.where(:cp_id => modified_product_ids[0]).first if modified_product_ids.any?
+        {:cert => product.certificate, :key => product.key}
+      end
+
```

Comment:
Are there cases where the product will be nil after line 83? I wasn't able to get it to fail at that line in my tests.